targets/TARGET_STM/sleep.c

```diff
@@ -40,17 +40,25 @@ extern void HAL_ResumeTick(void);
 
 void hal_sleep(void)
 {
+    // Disable IRQs
+    core_util_critical_section_enter();
```

Comment:
Presumably yes, but it depends on what these platforms are doing in `hal_sleep()`. If they are calling just a `__WFI` and have no other constraints, the there is no need for a critical section.